src/chttpd/src/chttpd_db.erl

```diff
@@ -889,7 +888,7 @@ send_all_docs_keys(Db, #mrargs{} = Args, UserCtx, VAcc0) ->
         _ -> Args#mrargs.doc_options
     end ++ [{user_ctx, UserCtx}],
     IncludeDocs = Args#mrargs.include_docs,
-    VAcc2 = lists:foldl(fun(DocId, Acc) ->
+    lists:foldl(fun(DocId, Acc) ->
         OpenOpts = [deleted | DocOpts],
         Row0 = case fabric2_db:open_doc(Db, DocId, OpenOpts) of
```

Comment:
This looks suspect. Is this code invoked for the _local docs namespace?

Reply:
Yeah it gets invoked for the _local_docs namespace as well. It seems to work since we get the total using the namespace and then we fetch individual docs with open_doc which also opens _local docs properly.

```
$ http 'http://adm:pass@localhost:15984/db/_local_docs?keys=["_local/ldoc1", "_design/ldoc2"]'

{
    "offset": null,
    "rows": [
        {
            "id": "_local/ldoc1",
            "key": "_local/ldoc1",
            "value": {
                "rev": "0-1"
            }
        },
        {
            "error": "not_found",
            "key": "_design/ldoc2"
        }
    ],
    "total_rows": 2
}
```

Another odd think about it is that with `keys` you can request non-local or non-design docs even if you go to _local_docs or _design_docs but you can that on master as well: 

(on master)
```
$ http 'http://adm:pass@localhost:15984/db/_local_docs?keys=["_local/l1", "_design/des1"]'

{
    "offset": null,
    "rows": [
        {
            "id": "_local/l1",
            "key": "_local/l1",
            "value": {
                "rev": "0-1"
            }
        },
        {
            "id": "_design/des1",
            "key": "_design/des1",
            "value": {
                "rev": "1-4b8a35d3f70a5962f86c6dd06ceb599c"
            }
        }
    ],
    "total_rows": null
}
```

Reply:
Cool beams.